### DISCUSSION TITLE
UPnP renderer has stopped working, etc.

### DISCUSSION THREAD

**Discussion by bnilsson11** [rern/rAudio#363] [Fixed Issues]
Some problems coming up during the last days (end of May, beginning of June)
Currently updated to 20210601.

1) I can no longer find the rAudio UPnP renderer on my network (it is Enabled in rAudio);
2) Wifi access point is always on after reboot, (I did turn it off):
3) Wifi is always enabled after reboot (I did disable it);
4) My NFS NAS is not mounted after reboot, it mounts just fine manually in System;

**Comment by bnilsson11** [comment]
Until the issue with the UPnP rendering is solved, I have to use HiFiBerryOS instead of rAudio.
rAudio is superior to HiFiBellyOS in many respects, but the HiFiBerryOS UPnP rendering is stable and working so I have to use that for the time being.

**Comment by rern** [comment]
Try today's update.
Some comparisons to share?

**Reply by bnilsson11** [reply]
Now the UPnP renderer is back on line on my network, and it is working fine.
Thanks!

**Comment by bnilsson11** [comment]
Hi,

I again have severe problems with the UPnP rendering function.
I mix the use of rAudio between playing from my NAS or local disk from the rAudio GUI, and streaming from Qobuz via BubbleUPnP using rAudio as UPnP renderer. For me it has been a perfect solution.

Now the BubbleUPnP with rAudio UPnP rendering is not working well. Same track is sometimes repeated, wrong track is played, etc.
When I watch the rAudio GUI while playing from BubbleUPnP I notice that the rAudio queue is not cleared when playing by UPnP rendering, and I guess this might be the problem. Tracks are coming and going in an unpredictable order in the rAudio playlist, unpredictable at least to me.

This is a classical example of who should take the blame when two software functions from different sources are not working well together. 

As a test, I created a SD with Raspberry Pi OS and installed upmpdcli and mpd, nothing else.
I connected a mpd client (M.A.P.D) to monitor what was going on, and I found the following:
1) BubbleUPnP was in perfect control, playing tracks by jumping in the playlist, adding, removing tracks, etc. was working without problems. The selected track selected in BubbleUPnP was always played correctly.
2) Looking into the M.A.P.D mpd client track queue while playing, there was only ONE track at any one time, not several.

Back in the old days, there was an option for the UPnP rendering feature, to set who would control te UPnP queue, Rune or the controller. This option is now removed, so it is not clear to me who is running the show.

Of course one could argue "well, use this setup then if it is working so well for you" but it is not that simple. Firstly, rAudio can use a "RPI hat", while my test upmpdcle/mpd setup cannot, only USB, and not selectable by any GUI. Secondly, as many are commenting, "after trying different alternatives, I always come back to Rune/rAudio". I think this is very true, the functionality of rAudio is outstanding. For me, inventing the wheel again by learning how to interface I2S and RPI hat to mpd is not an option.

I hope this issue can be fixed, so everything would go back to normal.

BN

**Reply by bnilsson11** [reply]
I noticed that after installing the latest rAudio-1 image (as of today 20211121) and upgrading ArchLinux, the UPnP rendering feature does not start at all.

`friendlyname = rAudio
ownqueue = 0
iconpath = /srv/http/assets/img/icon.png
onstart = /srv/http/bash/upmpdcli.sh
onplay = /srv/http/bash/cmd-pushstatus.sh
onpause = /srv/http/bash/cmd-pushstatus.sh
onstop = /srv/http/bash/cmd-pushstatus.sh

● upmpdcli.service - UPnP Renderer front-end to MPD
     Loaded: loaded (/etc/systemd/system/upmpdcli.service; enabled; vendor preset: disabled)
    Drop-In: /etc/systemd/system/upmpdcli.service.d
             └─override.conf
     Active: failed (Result: exit-code) since Sun 2021-11-21 11:09:28 UTC; 3s ago
    Process: 2291 ExecStart=/usr/bin/taskset -c 3 /usr/bin/upmpdcli -c /etc/upmpdcli.conf (code=killed, signal=TERM)
    Process: 2292 ExecStartPost=/srv/http/bash/cmd.sh upnpnice (code=exited, status=1/FAILURE)
   Main PID: 2291 (code=killed, signal=TERM)
        CPU: 77ms

Nov 21 11:09:28 rAudio taskset[2291]: :3:src/mediaserver/contentdirectory.cxx:187::ContentDirectory: not creating entry for deezer because neither deezeruser nor deezerautostart are defined in the configuration
Nov 21 11:09:28 rAudio taskset[2291]: :3:src/mediaserver/contentdirectory.cxx:187::ContentDirectory: not creating entry for hra because neither hrauser nor hraautostart are defined in the configuration
Nov 21 11:09:28 rAudio taskset[2291]: :3:src/mediaserver/contentdirectory.cxx:187::ContentDirectory: not creating entry for qobuz because neither qobuzuser nor qobuzautostart are defined in the configuration
Nov 21 11:09:28 rAudio taskset[2291]: :3:src/mediaserver/contentdirectory.cxx:187::ContentDirectory: not creating entry for spotify because neither spotifyuser nor spotifyautostart are defined in the configuration
Nov 21 11:09:28 rAudio taskset[2291]: :3:src/mediaserver/contentdirectory.cxx:187::ContentDirectory: not creating entry for tidal because neither tidaluser nor tidalautostart are defined in the configuration
Nov 21 11:09:28 rAudio taskset[2291]: :3:src/mediaserver/contentdirectory.cxx:187::ContentDirectory: not creating entry for upradios because neither upradiosuser nor upradiosautostart are defined in the configuration
Nov 21 11:09:28 rAudio taskset[2291]: :3:src/mediaserver/contentdirectory.cxx:187::ContentDirectory: not creating entry for uprcl because neither uprcluser nor uprclautostart are defined in the configuration
Nov 21 11:09:28 rAudio systemd[1]: upmpdcli.service: Control process exited, code=exited, status=1/FAILURE
Nov 21 11:09:28 rAudio systemd[1]: upmpdcli.service: Failed with result 'exit-code'.
Nov 21 11:09:28 rAudio systemd[1]: Failed to start UPnP Renderer front-end to MPD.`
BN

**Reply by bnilsson11** [reply]
I tried to flash the img again, this time I did NOT do the ArchLinux upgrade. The result is the same, the UPnP renderer cannot be started. Addons upgrade did not improve this.

/BN

**Reply by rern** [reply]
Try:
- `rm -rf /etc/systemd/system/upmpdcli.service.d` (missing from today's update)
- Enable UPnP again

**Reply by bnilsson11** [reply]
Now the UPnP renderer starts ok, thanks.
Did you make any other changes?

/BN

**Reply by bnilsson11** [reply]
I finally found out how to make the renderer behave.

In /etc/upmpdcli.conf:
change
ownqueue = 0
to 
ownqueue = 1

Now the renderer plays correctly what the DLNA control point sends to it.

Please feel free to explain what is going on, anyone who has the knowledge.
Also, explain the reason why the default setting is 0 and not 1.

/BN

**Reply by bnilsson11** [reply]
After having studied the subject some more, I would strongly suggest that upmpdcli ownqueue **should be set to 1 by default** in the rAudio distribution.

/BN

**Reply by bnilsson11** [reply]
The latest update 29211203 (or did "Clear playlist at start" appear earlier? I might have missed it...) fixed this issue.
Thanks a lot!

**Reply by elliotn** [reply]
I'm having erratic results with UPNP playback after changing input sources on my DAC. I'm using an SMSL SU-6 DAC which disconnects from USB when I change to another source. When I change back to USB, it reconnects to rAudio. I'm running "rAudio 1 • 20211203".

I was playing something from my NAS. Switched sources. Music stopped. Switched back to USB. Hit play, and this is what I see in the logs:

:2:src/mpdcli.cxx:237::MPDCli::eventloop: mpd_run_idle_mask returned 0
:2:src/mpdcli.cxx:237::MPDCli::eventloop: mpd_run_idle_mask returned 0
:2:src/mpdcli.cxx:331::mpd_run_play(m_conn) failed: Connection closed by the server

This is what my upmpdcli.conf file looks like (I commented out a few commands as cmd-pushstatus.sh does not exist):
friendlyname = star
ownqueue = 1
iconpath = /srv/http/assets/img/icon.png
onstart = /srv/http/bash/upmpdcli.sh
#zzz onplay = /srv/http/bash/cmd-pushstatus.sh
#zzz onpause = /srv/http/bash/cmd-pushstatus.sh
#zzz onstop = /srv/http/bash/cmd-pushstatus.sh

logfilename = /var/log/upmpdcli.log
loglevel = 2

----
Any thoughts?

**Reply by elliotn** [reply]
After enabling logging in mpd and upmpdcli and poking around the file system, my UPNP playback seems to be stable. This includes pause/play/stop behavior. Here's my upmpdcli.conf, in case it might help someone else:

---

friendlyname = star
ownqueue = 1
iconpath = /srv/http/assets/img/icon.png
onstart = /srv/http/bash/upmpdcli.sh

#zzz stuff i've added

onplay = /srv/http/bash/status.sh
onpause = /srv/http/bash/status.sh
onstop = /srv/http/bash/status.sh

logfilename = /var/log/upmpdcli.log
loglevel = 2

openhome = 0

**Reply by rern** [reply]
Thank you for finding this bug. The existing `/srv/http/bash/cmd-pushstatus.sh` lines have not been changed to `/srv/http/bash/status-push.sh`.
Next update will have the file fixed.

**Reply by elliotn** [reply]
Thank you @rern!!
